### PR TITLE
Evaluate promises & active bindings in env_get()

### LINKED ISSUE #554: env_print incorrectly labels binding as promise
``` r
library(rlang)

power1 <- function(exponent) {
  function(x) {
    x ^ exponent
  }
}

square <- power1(2)
square(2)
#> [1] 4

env <- get_env(square)
env_print(env)
#> <environment: 0x7ff5c3cf7988>
#>   parent: <environment: global>
#>   bindings:
#>    * exponent: <promise>

env$exponent
#> [1] 2
env_print(env)
#> <environment: 0x7ff5c3cf7988>
#>   parent: <environment: global>
#>   bindings:
#>    * exponent: <promise>
```

Created on 2018-07-03 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

### PR DESCRIPTION
**Pull request by hadley**
Fixes #554

### CODE REVIEW

File: src/internal/env-binding.c [hunk 1]
```diff
@@ -40,5 +40,9 @@ sexp* rlang_env_get(sexp* env, sexp* nm) {
     r_abort("Internal error: `rlang_env_get()` should have failed earlier");
   }
 
+  if (r_typeof(out) == r_type_promise || r_typeof(out) == r_type_function) {
```
Comment (lionel-) [inline src/internal/env-binding.c]: Won't the second alternative evaluate bindings to a function? I think active bindings have promise type.

Comment (hadley) [reply src/internal/env-binding.c]: Oooh good point; looks like I should be using this: https://github.com/wch/r-source/blob/496adb0085703362b2dee63537f0d89820f5afbb/src/include/Defn.h#L430

Comment (hadley) [reply src/internal/env-binding.c]: But you're right that it is a subclass of promise

Comment (hadley) [reply src/internal/env-binding.c]: Fixed now

**Pull request merged**